rotkehlchen/db/accounting_rules.py

```diff
@@ -336,3 +304,160 @@ def query_rules_and_serialize(
+    bindings_and_events_iterator = zip(bindings, related_events)
+    processed_events: dict[int, bool] = {}  # map of all the events related to the one requested to wether they will be processed in accounting or not  # noqa: E501
+    with db.conn.read_ctx() as cursor:
+        # index to keep the current event in the list of related events. It is used in the
+        # callbacks since we need to process events but we don't want to consume the current
+        # iterator
+        current_event_index = 0
+        for event_binding, event in bindings_and_events_iterator:
+            assert event.identifier is not None, 'We should only process events with identifiers'
+            if (cached_is_missing_rule := accountant.processable_events_cache.get(event.identifier)) is not None:  # noqa: E501
+                processed_events[event.identifier] = cached_is_missing_rule
+                current_event_index += 1
+                continue
+
+            if isinstance(event, EthStakingEvent) is True:
+                # staking events all have a process() function for accounting
+                processed_events[event.identifier] = False
+                current_event_index += 1
+                continue
+
+            # check the rule in the database
+            row = cursor.execute(query, event_binding).fetchone()
+            is_missing_rule = row[0] == 0
+            processed_events[event.identifier] = is_missing_rule
+
+            # check the next events to verify if this particular event will
+            # be processed by this one in a callback or treatment.
+            new_missing_accounting_rule = _events_to_consume(
+                cursor=cursor,
+                callbacks=callbacks,
+                events_iterator=bindings_and_events_iterator,
+                next_events=related_events[current_event_index + 1:],
+                event=event,
+                pot=accounting_pot,
+            )
+            if len(new_missing_accounting_rule) != 0:
+                current_event_index += len(new_missing_accounting_rule)
+                if is_missing_rule is True:  # we processed it in the callback so is not missing
+                    processed_events[event.identifier] = False
+
+                # update the current event in the cache
+                accountant.processable_events_cache.add(event.identifier, processed_events[event.identifier])  # noqa: E501
+
+                # update information about the new events
+                for processed_event_id in new_missing_accounting_rule:
+                    accountant.processable_events_cache.add(processed_event_id, False)
+                    processed_events[processed_event_id] = False
+
+    # get the list of wether events are processable for the events requested and not the requested
+    # plus the related to them
+    is_missing_rule_list = [processed_events.get(event.identifier, False) for event in events]  # type: ignore[arg-type]
+    return is_missing_rule_list
```

Comment:
Why not use the cache directly? Why do you need `processed_events` to exist at all?

```suggestion
    return [processed_events.get(event.identifier, False) for event in events]  # type: ignore[arg-type]
```